### DISCUSSION TITLE
Shelly EM backup firmware

### DISCUSSION THREAD

**Discussion by barbatrukko** [indomus/forum#1725] [Hardware & Firmware]
Salve, devo effettuare il backup del firmware di uno shelly EM ( o in alternativa sovrascriverlo) per via della password dell'interfaccia web che non viene rimossa dal factory reset.
Ho installato pyton , ho preso l'adattatore USB / seriale ma non riesco a connettermi: credo che sbaglio nel metterlo in program mode.

Lo shelly EM ha 6 contatti (minuscoli) più un tasto reset.
I contatti dovrebbero essere questi https://shelly.cloud/knowledge-base/devices/shelly-em/#pinout
1,4,5,6
Ma non è chiaro come entrare in program mode.

Qualcuno di voi lo ha fatto?

**Comment by NdK73** [comment]
Normale programmazione ESP::
- RST a GND
- GPIO0 a GND
- RST scollegato (alto, float)
- GPIO0 float

Normalmente le schede di programmazione lo gestiscono automaticamente con le linee RTS e DTR della seriale (se cerchi "esp programmer schematic" ne trovi a volontà, e ti semplifica anche l'uso di ESPTOOL).

**Comment by barbatrukko** [comment]
Dunque poi ho risolto così:
Collegati stabilmente
1-> GND
4-> 3,3V
5,66 -> TX e RX incrociati con il programmatore...

prima di connettere USB il pin 2 (GPIO) l'ho messo in contatto con pin1 GND con un filo volante.
Inserita la porta usb ho tolto da GND il pin2 e si è attivato il PROGRAM MODE.
A questo punto con esptools php ha scaricato il firmware presente sul dispositivi.
